pandas/core/sparse/array.py

```diff
@@ -704,41 +893,154 @@ def mean(self, axis=0, *args, **kwargs):
             nsparse = self.sp_index.ngaps
             return (sp_sum + self.fill_value * nsparse) / (ct + nsparse)
 
-    def value_counts(self, dropna=True):
-        """
-        Returns a Series containing counts of unique values.
+    def transpose(self, *axes):
+        """Returns the SparseArray."""
+        return self
 
-        Parameters
-        ----------
-        dropna : boolean, default True
-            Don't include counts of NaN, even if NaN is in sp_values.
+    @property
+    def T(self):
+        """Returns the SparseArray."""
+        return self
+
+    # ------------------------------------------------------------------------
+    # Ufuncs
+    # ------------------------------------------------------------------------
+    def __abs__(self):
+        return np.abs(self)
+
+    def __array_ufunc__(self, ufunc, method, *inputs, **kwargs):
```

Comment:
`__array_ufunc__` is new in NumPy 1.13, which is older than our minimum supported version.

If anyone is interested in making this compatible with older numpy's I'd appreciate help here.

Reply:
When would you recommend raising an error? When the sparsearray is created, or when a ufunc is called? Either is possible. I would say when created.

Reply:
Ahh, I suppose because of `get_dummies` it would have to be when you call a ufunc... That's unfortunate.